### ISSUE TITLE
Whirligig (451650)

### ISSUE THREAD

**Issue by JulianGro** [ValveSoftware/Proton#2935]
# Compatibility Report
- Name of the app with compatibility issues: Whirligig
- Steam AppID of the app: 451650

## System Information
- GPU: GTX 1080
- Driver/LLVM version: 430.40
- Kernel version: 5.0.0
- Link to full system information report: https://gist.github.com/jug007/062419bcb740176ddad2af8aea917273
- Proton version: 4.11-1

## I confirm:
- [Yes] that I haven't found an existing compatibility report for this game.
- [Yes] that I have checked whether there are updates for my system available.

 (Will add this later)

## Symptoms
No video paths other than VideoLAN work.
VideoLAN has very bad performance compared to Windows.

## Reproduction
-Start Whirligig
-Try playing a video with and without VideoLAN as video path selected

**Comment by yaomtc** [comment]
The latest versions stopped Whirligig from being usable for me, I had to revert to 4.7462. It looks like you're not having my issue though? (Vive controllers not appearing, UI went down out of reach of the mouse cursor)

EDIT: This is with a Radeon RX 580 (Mesa), Linux kernel 5.2.3, latest SteamVR beta + Proton 4.2

**Comment by JulianGro** [comment]
@yaomtc try the new proton 4.11-1
I personally am using the Valve Index without controllers in Whirligig. (The controllers were not even on when I used it yesterday) If the ui is just out of reach you should be able to shift your playspace with OpenVR-AdvancedSettings. Or there is probably a button combination for resetting the view.

**Comment by yaomtc** [comment]
Thanks, the issue is gone in 4.11!

**Comment by Dexter9313** [comment]
For me it's not only the videos that have bad performance, it's the whole program ! Navigating through the menus with Whirligig freshly installed and no video loaded is already painful or impossible. What is your experience about this ?

**Comment by JulianGro** [comment]
For me personally, the program itself runs as good as on Windows. "Only" video playback is slow.

**Comment by Dexter9313** [comment]
I just tested on a monster rig and indeed it doesn't have any issue without video playback... My i5-4690K may be the problem.

**Comment by yaomtc** [comment]
I can no longer access anything in my home folder from Whirligig or from Skybox (#3896). Here's my launch options for each:

`STEAM_COMPAT_MOUNTS=/home/myusername/:/mnt/WD1TBlue/ %command%`

I figure this has to do with the [sandboxing of Pressure Vessel](https://gitlab.steamos.cloud/steamrt/steam-runtime-tools/-/blob/main/pressure-vessel/wrap.1.md) but I'm not sure... this used to work but it doesn't anymore.

**Comment by kisak-valve** [comment]
Hello @yaomtc, that's an issue to be pondered with the Steam Linux Runtime - Soldier container environment. Please open a new issue report in https://github.com/ValveSoftware/steam-runtime/.

**Comment by Crystal4276** [comment]
Hello,
I'm wondering if proton can handle video acceleration for decoding? 
I noticed that whatever the video path i'm choosing in whirligig, my CPU is used instead of GPU.

EDIT: actually the only usable video path is direct show. The other ones either doesn't work or lag crazy. 
But directshow still hit the cpu instead of gpu. 

My GPU is an RTX 2080.
Driver Version: 525.60.11

**Comment by yaomtc** [comment]
I don't think so, this is the latest news I could find:

https://www.phoronix.com/news/MTY2MTY

Skybox has better performance for video, I would recommend using that or vr-video-player: https://git.dec05eba.com/vr-video-player/about/

**Comment by Crystal4276** [comment]
> I don't think so, this is the latest news I could find

I do think so. 
By monitoring resource utilization, i can see my gpu staying close to 2-5% while the cpu jump to 50% on a 16 cores,

> Skybox has better performance for video, I would recommend using that or vr-video-player: https://git.dec05eba.com/vr-video-player/about/

I appreciate the mention of alternatives, but I think you are missing the point of what issues on Github are for. 
In this particular case, it's about having Whirligig running properly using Proton.

**Comment by yaomtc** [comment]
What? You asked if Wine has support for hardware accelerated video. I told you it doesn't, as is demonstrated by your GPU utilization... There's no indication that there's work being done to fix that, so I gave you a workaround.

**Comment by Crystal4276** [comment]
>What? You asked if Wine has support for hardware accelerated video. I told you it doesn't, as is demonstrated by your GPU utilization... There's no indication that there's work being done to fix that, so I gave you a workaround.

I don't know why i understood your previous post completely sideways. My bad...

Anyway, i hope this recent implementation of video acceleration in Vulkan would be useful for Proton/Wine and at the end Whirligig:
https://www.phoronix.com/news/Vulkan-Video-1.0

**Comment by yaomtc** [comment]
Looks like Unity (engine used for Whirligig) still only supports DX11 on Windows, Vulkan support is only on Android (like the Quest) 

https://issuetracker.unity3d.com/issues/xr-vulkan-openxr-open-xr-mock-hmd-on-windows-using-vulkan-renders-only-a-black-screen

https://docs.unity3d.com/Packages/com.unity.xr.openxr@1.6/manual/index.html

**Comment by Crystal4276** [comment]
> Looks like Unity (engine used for Whirligig) still only supports DX11 on Windows

Yes, but isn't it the role of DXVK (and by extension Proton ) to translate DX11 calls to Vulkan ?

**Comment by yaomtc** [comment]
DirectX 11 doesn't seem to be relevant to video playback, that's the job of DXVA (Media Foundation integrates it), which Wine does implement (translates to VA-API) but I'm not sure if it's working with Whirligig, not sure how I'd troubleshoot that...

**Comment by yaomtc** [comment]
Does not launch with Proton 8. Proton 7 works.

**Comment by kisak-valve** [comment]
Hello @yaomtc, please add [`PROTON_LOG=1 %command%`](https://github.com/ValveSoftware/Proton#runtime-config-options) to the game's launch options, reproduce the regression, and attach the generated $HOME/steam-$APPID.log to this issue report as a file. (Proton logs compress well if needed.) Also, please copy your system information from Steam (`Steam` -> `Help` -> `System Information`) and put it in a [gist](https://gist.github.com/), then include a link to the gist in this issue report.

**Comment by yaomtc** [comment]
https://gist.github.com/yaomtc/4fb6a5e6924936077adf1cddf77d67b1

Unlike with [SKYBOX](https://github.com/ValveSoftware/Proton/issues/3896), Whirligig only seems to crash on launch sometimes. Not every time. 

[steam-451650.log](https://github.com/ValveSoftware/Proton/files/11427491/steam-451650.log)

**Comment by extraymond** [comment]
> 

For me it crashes if the previously launched app/game, cough cough skybox, crashed silently.
After restarting only steamvr, I'm able to launch whirligig without problem, even with proton 8.

**Comment by Kagukara** [comment]
Using DirectShow, when the video loads and starts playing, if you pause it will not skip at first. Once you play the video and then pause again it will jump about 5 seconds. Once it starts skipping it seems to skip greater amounts. But the video its self doesn't skip and will play from where you paused. This only shows on the UI, so when you get to the end of the "video" according to the UI you will not be at the end of the video, cutting it of short.

Here is the PROTON_LOG=1 for [GE-Proton8-4](https://gist.githubusercontent.com/Kagukara/fdd0fb2d65a7382ea0f446d9277fb0ac/raw/b6a72c58f1c9f054968c5f79a5a2d5da3f24eb09/steam-451650-GE-Proton8-4.log) and [proton-experimental](https://gist.githubusercontent.com/Kagukara/fdd0fb2d65a7382ea0f446d9277fb0ac/raw/b6a72c58f1c9f054968c5f79a5a2d5da3f24eb09/steam-451650-proton-experimental.log) via [gist](https://gist.github.com/Kagukara/fdd0fb2d65a7382ea0f446d9277fb0ac). I paused at 00:05 which didn't skip, 00:10 which skipped to 00:15 and 00:20 which skipped to 00:30. This is consistent every time I open Whirligig for every video.

The developer was unable to replicate this on windows. This happens on a fresh install, with and without the codecs installed. Programs/Codecs I've installed in the Proton prefix:

- K-Lite_Codec_Pack_1766_Standard/LAVFilters-0.77.2-Installer
- vlc-3.0.18-win64.exe

EDIT: Found a fix, posted it to the Steam community discussions: [https://steamcommunity.com/app/451650/discussions/0/3809529690301811076/](https://steamcommunity.com/app/451650/discussions/0/3809529690301811076/)
